executor/Dockerfile.build

```diff
@@ -0,0 +1,13 @@
+FROM python:3.5
+
+RUN pip install pyinstaller
+
+RUN mkdir /opt/cook
+WORKDIR /opt/cook
+
+COPY requirements.txt /opt/cook/
+RUN pip install -r requirements.txt
+
+COPY . /opt/cook
+
+CMD ["pyinstaller", "-F", "-n", "cook-executor", "-p", "cook", "cook/__main__.py"]
```

Comment:
Copying over comments from the original PR (PR #308):

@dposada 2 days ago
> Why does this need to happen in a container?

@shamsimam a day ago
> The executable needs to be compiled using the same architecture as where it will be run - on the agents inside docker.